src/main/java/io/r2dbc/postgresql/client/ReactorNettyClient.java

```diff
@@ -111,6 +113,18 @@
             sink.next(message);
         });
 
+    private class EnsureSubscribersCompleteChannelHandler extends ChannelDuplexHandler {
+
+        @Override
+        public void channelUnregistered(ChannelHandlerContext ctx) throws Exception {
+            super.channelUnregistered(ctx);
+            requestProcessor.onComplete();
+            for (MonoSink<Flux<BackendMessage>> responseReceiver = responseReceivers.poll(); responseReceiver != null; ) {
```

Comment:
Note that this still needed `responseReceivers.poll()` as the third argument to the for loop to go to the next element in the `Queue`.  I'll fix it in a following commit.

Reply:
Oh，forgot that，thanks！